### ISSUE TITLE
Crash issue in LPMessagingSDK iOS

### ISSUE THREAD

**Issue by tameshwari** [LivePersonInc/developers-community#139]
I have integrated LPMessagingSDK in my iOS application. When I start chat for the first time. It works as expected. But If I kill the application and relaunch it again. It is crashing with below crash log. 

Please help me out. 

Crashed: SQLQueue 0x10675e690 for Consumer.sqlite
0  (Missing)                      0x104e1af48 (Missing)
1  (Missing)                      0x104e1a71c (Missing)
2  (Missing)                      0x104e1a748 (Missing)
3  (Missing)                      0x104df0498 (Missing)
4  (Missing)                      0x104df05b4 (Missing)
5  libdyld.dylib                  0x1847e47a8 dyld_stub_binder + 60
6  LPInfra                        0x105371908 _T07LPInfra17OverflowingBufferC7sendAll33_3B6C015BA679803417E3C34F44AA327CLL10Foundation4DataVAH4data_tFTf4gXn_n + 788
7  LPInfra                        0x105374fd8 _T07LPInfra17OverflowingBufferC6update10Foundation4DataVAG04withF0_tFTf4gXn_n + 124
8  LPInfra                        0x105379d40 _T07LPInfra11LPRNCryptorO11DecryptorV3C6update10Foundation4DataVAI04withG0_tKFTf4gXn_n + 5172
9  LPInfra                        0x10537008c _T07LPInfra11LPRNCryptorO11DecryptorV3CAA0B4TypeA2aFP6update10Foundation4DataVAK04withH0_tKFTW + 72
10 LPInfra                        0x10537b9c0 _T07LPInfra11LPRNCryptorO9DecryptorC6update10Foundation4DataVAI04withF0_tKFTf4xn_n + 3928
11 LPInfra                        0x10537c588 _T07LPInfra15LPRNCryptorTypePAAE7oneshot33_3B6C015BA679803417E3C34F44AA327CLL10Foundation4DataVAH4data_tKFAA0B0O9DecryptorC_Tg5Tf4gXn_n + 88
12 LPInfra                        0x10542c1bc _T07LPInfra17EncryptionManagerC11decryptData10Foundation0E0VSgAGFTf4gXd_n + 420
13 LPInfra                        0x10534f304 _T07LPInfra27StringEncryptionTransformerC23reverseTransformedValueypSgAEF + 1180
14 LPInfra                        0x10534f558 _T07LPInfra27StringEncryptionTransformerC16transformedValueypSgAEFToTm + 104
15 CoreData                       0x1877bfa40 _prepareResultsFromResultSet + 3588
16 CoreData                       0x1877c046c _newFetchedRowsForFetchPlan_ST + 1440
17 CoreData                       0x1877b320c _executeFetchRequest + 68
18 CoreData                       0x1877adaf0 -[NSSQLFetchRequestContext executeRequestCore:] + 36
19 CoreData                       0x1876141d4 -[NSSQLStoreRequestContext executeRequestUsingConnection:] + 248
20 CoreData                       0x187696118 __52-[NSSQLDefaultConnectionManager handleStoreRequest:]_block_invoke + 84
21 libdispatch.dylib              0x18477ea14 _dispatch_client_callout + 16
22 libdispatch.dylib              0x1847bbd4c _dispatch_queue_barrier_sync_invoke_and_complete + 56
23 CoreData                       0x187696014 -[NSSQLDefaultConnectionManager handleStoreRequest:] + 256
24 CoreData                       0x187786524 -[NSSQLCoreDispatchManager routeStoreRequest:] + 264
25 CoreData                       0x1876c7fdc -[NSSQLCore dispatchRequest:withRetries:] + 236
26 CoreData                       0x1876c438c -[NSSQLCore processFetchRequest:inContext:] + 108
27 CoreData                       0x1875a6bb4 -[NSSQLCore executeRequest:withContext:error:] + 496
28 CoreData                       0x1876a31fc __65-[NSPersistentStoreCoordinator executeRequest:withContext:error:]_block_invoke + 1340
29 CoreData                       0x18769bcc4 -[NSPersistentStoreCoordinator _routeHeavyweightBlock:] + 268
30 CoreData                       0x1875a6680 -[NSPersistentStoreCoordinator executeRequest:withContext:error:] + 416
31 CoreData                       0x1875a515c -[NSManagedObjectContext executeFetchRequest:error:] + 580
32 LPInfra                        0x105420f78 _T07LPInfra11DataManagerC13queryEntitiesSayxGSS10entityName_So11NSPredicateCSg9predicateSo22NSManagedObjectContextCSg7contextSiSg14fetchBatchSizetSo0jK0CRbzlFAA12LPFileEntityC_Tg5Tf4gXggnn_nTm + 348
33 LPInfra                        0x105421410 _T07LPInfra11DataManagerC13getEntityByIdxSgxm_SS2idSo22NSManagedObjectContextCSg7contextSiSg14fetchBatchSizetSo0iJ0CRbzlFAA07LPBrandE0C_Tg5Tf4ngXgnn_nTm + 560
34 LPInfra                        0x1054211d4 _T07LPInfra11DataManagerC13getEntityByIdxSgxm_SS2idSo22NSManagedObjectContextCSg7contextSiSg14fetchBatchSizetSo0iJ0CRbzlFAA06LPUserE0C_Tg5Tf4ngXgnn_n + 44
35 LPInfra                        0x105447464 _T07LPInfra11UserManagerC03getB12FromDatabaseAA12LPUserEntityCSgSS6userID_tFTf4gXn_n + 708
36 LPInfra                        0x105449e80 _T07LPInfra11UserManagerC02myB2IDSSSgfWTf4dn_n + 120
37 LPInfra                        0x105443ac8 _T07LPInfra11UserManagerC06loadMyB14IdFromSavedJWTyyF + 776
38 LPInfra                        0x1054436c8 _T07LPInfra11UserManagerCACyc33_89C6461B4946C110E6BB94DA86C0E248Llfc + 628
39 LPInfra                        0x1054436f4 _T07LPInfra11UserManagerCACyc33_89C6461B4946C110E6BB94DA86C0E248LlfcTo + 20
40 LPInfra                        0x105443740 globalinit_33_89C6461B4946C110E6BB94DA86C0E248_func248 + 64
41 libdispatch.dylib              0x18477ea14 _dispatch_client_callout + 16
42 libdispatch.dylib              0x1847b1178 dispatch_once_f$VARIANT$armv81 + 60
43 LPInfra                        0x105388668 _T07LPInfra13PusherManagerCACyc33_98E134C78BDB863AB8A74F2F6C234BF4Llfc + 296
44 LPInfra                        0x1053887bc _T07LPInfra13PusherManagerCACyc33_98E134C78BDB863AB8A74F2F6C234BF4LlfcTo + 20
45 LPInfra                        0x1053884c0 globalinit_33_98E134C78BDB863AB8A74F2F6C234BF4_func169 + 64
46 libdispatch.dylib              0x18477ea14 _dispatch_client_callout + 16
47 libdispatch.dylib              0x1847b1178 dispatch_once_f$VARIANT$armv81 + 60
48 LPInfra                        0x1053d42b0 _T07LPInfra0A6FacadeC14setPusherTokeny10Foundation4DataV5token_SSSg17alternateBundleIDAA0E4TypeO0hL0tFZ + 96
49 LPMessagingSDK                 0x1058a04a8 _T014LPMessagingSDKAAC25registerPushNotificationsy10Foundation4DataV5token_AA0A23SDKNotificationDelegate_pSg012notificationJ0SSSg17alternateBundleIDtFTo + 208
50 myApp                        0x104a0e634 -[AppDelegate application:didRegisterForRemoteNotificationsWithDeviceToken:] (AppDelegate.m:667)
51 libdispatch.dylib              0x18477ea54 _dispatch_call_block_and_release + 24
52 libdispatch.dylib              0x18477ea14 _dispatch_client_callout + 16
53 libdispatch.dylib              0x1847bfc80 _dispatch_main_queue_callback_4CF$VARIANT$armv81 + 968
54 CoreFoundation                 0x184da7344 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
55 CoreFoundation                 0x184da4f20 __CFRunLoopRun + 2012
56 CoreFoundation                 0x184cc4c58 CFRunLoopRunSpecific + 436
57 GraphicsServices               0x186b70f84 GSEventRunModal + 100
58 UIKit                          0x18e41d5c4 UIApplicationMain + 236
59 myApp                        0x104a0bb44 main (main.m:16)
60 libdyld.dylib                  0x1847e456c start + 4

**Comment by dvillacis94** [comment]
Hi @tameshwari,

Which version of the SDK are you using?

Are you handling the SDK on the AppDelegate? Could you share those lines?

**Comment by tameshwari** [comment]
Hi,

I am using LPMessagingSDK 2.8.0.  I have proper setup in Appdelegate file. 
I am facing this issue only If I try to relaunch my application.

**Comment by tameshwari** [comment]
Hi @dvillacis94,

Is there any issue related to APNS certificate ?  
Previously It was working as expected. Only after changing APNS certificate, I am facing this issue.

**Comment by shaymLP** [comment]
Hi @tameshwari, did you performed any downgrade to Liveperson SDK?
Please make sure you follow our step by step guide while using push notifications: 
https://developers.liveperson.com/consumer-experience-ios-sdk-sampleapp.html

**Comment by shaymLP** [comment]
@tameshwari, As this was answered 5 days ago - This issue will be closed. Do not hesitate to open a new issue if you have any further questions.
